test/e2e/bootstrapper_test.go

```diff
@@ -35,15 +39,30 @@ func TestBootstrapper(t *testing.T) {
 	// TODO: Consider doing the same with kubelet. We can either provide our own or download it from the internet
 	// 		 if we choose to download it, we will have to compare expected vs actual SHA hashes for security reasons
 	ensureIgnitionFileExists(t, ignitionFilePath)
+
+	// Stop the kubelet service so we can remove the kubelet logfile
 	wmcb, err := bootstrapper.NewWinNodeBootstrapper(installDir, ignitionFilePath, kubeletPath)
 	assert.Nilf(t, err, "Could not create WinNodeBootstrapper: %s", err)
+	wmcb.StopAndRemoveServices()
+	wmcb.Disconnect()
+	time.Sleep(5 * time.Second)
+	// Remove the kubelet logfile, so that when we parse it, we are looking at the current run only
+	removeFileIfExists(t, kubeletLogPath)
+
 	// Run the bootstrapper, which will start the kubelet service
+	wmcb, err = bootstrapper.NewWinNodeBootstrapper(installDir, ignitionFilePath, kubeletPath)
+	assert.Nilf(t, err, "Could not create WinNodeBootstrapper: %s", err)
 	err = wmcb.Run()
 	assert.Nilf(t, err, "Could not run bootstrapper: %s", err)
 	err = wmcb.Disconnect()
 	assert.Nilf(t, err, "Could not disconnect from windows svc API: %s", err)
 	time.Sleep(2 * time.Second)
+
 	assert.Truef(t, svcRunning(t, bootstrapper.KubeletServiceName), "The kubelet service is not running")
+	// Wait for kubelet log to be populated
+	time.Sleep(5 * time.Second)
```

Comment:
Why 5 Second interval again?

Reply:
To wait for the kubelet log to be populated